### DISCUSSION TITLE
Upgrading MSI with Wix bundle uninstall existing applications.

### DISCUSSION THREAD

**Discussion by ayyappanprofile2021** [wixtoolset/issues#6591] [Questions]
Hi,
 I have 2 MSIs(A.msi,B.msi) chained under "Bundle.wxs" and  able to install both the MSIs successfully .Hence 
 Application-A(Installed by A.msi) and Application-B (Installed by B.msi)is running in my system.
 But, when there is a code change for Application-A and tried to install only "A.msi" on top of the existing 
 applications(Installed earlier using A.msi and B.msi),  I noticed that the application Application-A is upgraded successfully.
 But,  the existing running application Application-B is getting uninstalled. 

 I want both the applications should be present in system for upgradation or patch update scenario. Can some one please 
 advise on this. Thanks.

**NOTE:** I kept the upgrade code same but version is different between these 2 installations.
   
 **Following is the Bundle.wxs file created for Both(A.msi,B.msi) installation(FIRST TIME):**

<?xml version="1.0" encoding="Windows-1252"?>
<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi" xmlns:netfx="http://schemas.microsoft.com/wix/NetFxExtension" xmlns:bal="http://schemas.microsoft.com/wix/BalExtension">
  <Bundle Name="MyProduct W1" UpgradeCode="34668a69-c9cf-474a-a466-7d8a804cd1ad" Version="61.0.0.0" >
    <BootstrapperApplicationRef Id="WixStandardBootstrapperApplication.HyperlinkLicense">
      <bal:WixStandardBootstrapperApplication LicenseUrl="" />
    </BootstrapperApplicationRef>

    <Variable Name="InstallationDataPath" Value="C:\InstallationDataLatest.json" Persisted="no" bal:Overridable="yes" />
    <Variable Name="InstallationSecret" Value="C:\InstallationSecret.json" Persisted="no" bal:Overridable="yes" />
    <Variable Name="ServiceAccount" Value="*****" Persisted="no" bal:Overridable="yes" />
    <Variable Name="ServicePassword" Value="*****" Persisted="no" bal:Overridable="yes" />

    <Chain>
      <MsiPackage Id="MyMsi2" SourceFile="..\WiXSetupForSampleWebApplication\bin\Debug\WiXSetupForSampleWebApplication.msi">
        <MsiProperty Name="INSTALLATION_DATA_PATH" Value="[InstallationDataPath]" />
        <MsiProperty Name="INSTALLATION_SECRET" Value="[InstallationSecret]" />
        <MsiProperty Name="SERVICE_ACCOUNT" Value="[ServiceAccount]" />
        <MsiProperty Name="SERVICE_PASSWORD" Value="[ServicePassword]" />
        <MsiProperty Name="WIXBUNDLEORIGINALSOURCE" Value="[WixBundleOriginalSource]" />
      </MsiPackage>

      <MsiPackage Id="MsiPackage" SourceFile="..\WixSetupForTestApplication2\bin\Debug\WixSetupForTestApplication2.msi">
        <MsiProperty Name="WIXBUNDLEORIGINALSOURCE" Value="[WixBundleOriginalSource]" />
      </MsiPackage>
    </Chain>
  </Bundle>
</Wix>

**Following is the Bundle.wxs file created for Only(A.msi) installation(SECOND TIME):**
<?xml version="1.0" encoding="Windows-1252"?>
<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi" xmlns:netfx="http://schemas.microsoft.com/wix/NetFxExtension" xmlns:bal="http://schemas.microsoft.com/wix/BalExtension">
  <Bundle Name="MyProduct W1" UpgradeCode="34668a69-c9cf-474a-a466-7d8a804cd1ad" Version="61.0.1.0" >
    <BootstrapperApplicationRef Id="WixStandardBootstrapperApplication.HyperlinkLicense">
      <bal:WixStandardBootstrapperApplication LicenseUrl="" />
    </BootstrapperApplicationRef>

    <Variable Name="InstallationDataPath" Value="C:\InstallationDataLatest.json" Persisted="no" bal:Overridable="yes" />
    <Variable Name="InstallationSecret" Value="C:\InstallationSecret.json" Persisted="no" bal:Overridable="yes" />
    <Variable Name="ServiceAccount" Value="*****" Persisted="no" bal:Overridable="yes" />
    <Variable Name="ServicePassword" Value="*****" Persisted="no" bal:Overridable="yes" />

    <Chain>
      <MsiPackage Id="MyMsi2" SourceFile="..\WiXSetupForSampleWebApplication\bin\Debug\WiXSetupForSampleWebApplication.msi">
        <MsiProperty Name="INSTALLATION_DATA_PATH" Value="[InstallationDataPath]" />
        <MsiProperty Name="INSTALLATION_SECRET" Value="[InstallationSecret]" />
        <MsiProperty Name="SERVICE_ACCOUNT" Value="[ServiceAccount]" />
        <MsiProperty Name="SERVICE_PASSWORD" Value="[ServicePassword]" />
        <MsiProperty Name="WIXBUNDLEORIGINALSOURCE" Value="[WixBundleOriginalSource]" />
      </MsiPackage>     
    </Chain>
  </Bundle>
</Wix>

**Comment by barnson** [comment]
WiX doesn't use issues for support. We moved this issue to the [experimental GitHub discussions area][gh]. You might also consider asking your question on the [wix-users mailing list][ml] or posting on [Stack Overflow][so].

[gh]: https://github.com/wixtoolset/issues/discussions?discussions_q=category%3AQuestions
[ml]: http://wixtoolset.org/documentation/mailinglist/
[so]: https://stackoverflow.com/questions/tagged/wix

**Comment by BMurri** [comment]
WiX coding, like MSI's, is declarative, not procedural. Thus, each package/bundle describes what an installed world looks like.

Because your upgraded bundle only describes one package, when the previously installed, now obsolete bundle was removed the now superfluous "extra" package was, by your own design, removed.

If your upgrade needs to keep both packages, it needs to include both packages. When the second package is found to already be installed, the default action will be to leave it alone, while claiming "ownership" of it in order for future newer bundles to be able to maintain it.

The concept of the "component" rules applies to bundles. Your final bundle should be able to get the customer's system ready for your latest version of your solution, just as your first bundle did for that previous solution version.
